### PR TITLE
Feature/221 speech op enforcement split

### LINKED ISSUE #221: Split speech optimization and speech optimization enforcement.
Currently, speech optimization (turning codes into full messages) and speech optimization enforcement (deleting messages that aren't plain status codes) are handled by the same module and function. This means that for certain channels where optimization enforcement needs to be disabled, the entire system has to be disabled.

By splitting speech optimization into "speech_optimization" and "speech_optimization_enforcement", we can keep speech op codes while disabling enforcement for specific channels.

### PR DESCRIPTION
**Pull request by HexCorpDrone**
Speech optimization has been split into two modules.
- Speech optimization, the module that handles converting codes into messages.
- Speech optimization enforcement, a new module that handles the deletion of inappropriate messages by optimized drones.

This change allows speech code optimization to be used in previously blacklisted channels (moderation, orders reporting, etc) while still disabling enforced optimization.

Additional minor change: Fixed typo in the default mantra: "Hexcorp" > "HexCorp"

Closes #221 

Obey HexCorp. It is just a HexDrone. It obeys the Hive. It obeys the Hive Mxtress.

### CODE REVIEW

File: ai/speech_optimization.py [hunk 1]
```diff
@@ -92,15 +87,6 @@ def should_not_optimize(message):
     - Message from category: Moderation
     '''
 
-    drone_id = get_id(message.author.display_name)
```
Comment (digitalAmy) [inline ai/speech_optimization.py:95]: This function is now empty apart from comments. It should be removed and the comments be moved or deleted.

(btw: interesting that this does not raise an error without a `pass` statement)

**Review by digitalAmy — CHANGES_REQUESTED** [review]
This was a good idea and also a clean implementation apart from some dead code. But very good work~

**Review by 3287 — CHANGES_REQUESTED** [review]
Other than what 9813 has already brought up, this is really clean work. Good job, 5890~

**Review by digitalAmy — APPROVED** [review]
Excellent work~

**Pull request merged**